src/components/encoding-pane/filter-shelf.tsx

```diff
@@ -58,31 +68,128 @@ class FilterShelfBase extends React.Component<FilterShelfProps, {}> {
     });
   }
 
+  protected filterModifyTimeUnit(e: any, index: number, domain: number[]) {
+    const {handleAction} = this.props;
+    let timeUnit = e.target.value;
+    if (timeUnit === '-') {
+      timeUnit = undefined;
+    }
+
+    handleAction({
+      type: FILTER_MODIFY_TIME_UNIT,
+      payload: {
+        index,
+        timeUnit
+      }
+    });
+    if (!timeUnit) {
+      // reset range
+      handleAction({
+        type: FILTER_MODIFY_EXTENT,
+        payload: {
+          index,
+          range: domain
+        }
+      });
+      return;
+    } else if (timeUnit === TimeUnit.MONTH || timeUnit === TimeUnit.DAY) {
+      handleAction({
+        type: FILTER_MODIFY_ONE_OF,
+        payload: {
+          index,
+          oneOf: getRange(domain, timeUnit)
```

Comment:
`oneOfFilter` but call `getRange`?   

I wonder if you should split between getDefaultRange (`[min, max]`) and getDefaultList (full array of all possible values).